src/foam/dao/AbstractDAO.js

```diff
@@ -284,6 +284,17 @@ foam.CLASS({
   ],
 
   methods: [
+    function find(key) {
+      if ( this.of && this.of.isInstance(key) ) {
+        if ( this.predicate.f(key) ) return this.delegate.find(key);
```

Comment:
I'm not a fan of this style of if/else, also if both branches are returning anyway, just do:
return this.predicate.f(key) ? this.delegate.find(key) : Promose.resolve(null);